### PR TITLE
Refactoring: Move Linux distribution/ files to linux/

### PR DESCRIPTION
**Pull request by ann0see**
**Short description of changes**

Moves Linux distribution files to linux/ folder and icons to src/res. We might even be able to reuse some of the images now.

CHANGELOG: Refactoring: Moved around files to clean up the repository

**Context: Fixes an issue?**

Related to: https://github.com/jamulussoftware/jamulus/pull/2601

**Does this change need documentation? What needs to be documented and how?**

No.

**Status of this Pull Request**

To be tested on Raspberry Pi with raspijamulus

**What is missing until this pull request can be merged?**

Review and decision on the weather we rename deb-mainicon.* to deb-jamulus.* since that's describing more what the file actually is. However that would break the naming style of Windows and macOS icons.

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want (deb installs and displays images fine locally, Raspi not tested)
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/master/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

File: Jamulus.pro [hunk 1]
```diff
@@ -683,11 +683,11 @@ DISTFILES += ChangeLog \
...
     CONTRIBUTING.md \
     README.md \
-    distributions/jamulus.desktop.in \
-    distributions/jamulus-server.desktop.in \
-    distributions/jamulus.png \
-    distributions/jamulus.svg \
-    distributions/jamulus-server.svg \
+    linux/jamulus.desktop.in \
+    linux/jamulus-server.desktop.in \
+    src/res/deb-mainicon.png \
```
Comment (pljones) [inline Jamulus.pro:688]: Why `deb-`?  These would be for [any freedesktop.org distribution](https://www.freedesktop.org/wiki/Specifications/desktop-entry-spec/), I think.

Comment (ann0see) [reply Jamulus.pro:688]: Since we have mac- win- as prefix and we mainly support debian. I'm also not convinced with the renaming at all.

Comment (ann0see) [reply Jamulus.pro:688]: I'd prefer jamulus.png jamulus.svg and jamulus-server.svg since these files are not necessarily linux only.

**Review by pljones — APPROVED** [review]
All **looks** good, not tried it out though.

**Comment by ann0see** [comment]
Yes, this still needs some investigation.

**Comment by ann0see** [comment]
I think I know what happened. The code is NOT ok.

File: Jamulus.pro [hunk 2]
```diff
@@ -326,29 +326,29 @@ win32 {
...
-        icons.files = distributions/jamulus.png
+        icons.files = src/res/io.jamulus.jamulus.png
 
         isEmpty(ICONSDIR_SVG) {
             ICONSDIR_SVG = share/icons/hicolor/scalable/apps/
         }
         ICONSDIR_SVG = $$absolute_path($$ICONSDIR_SVG, $$PREFIX)
         icons_svg.path = $$ICONSDIR_SVG
-        icons_svg.files = distributions/jamulus.svg distributions/jamulus-server.svg
+        icons_svg.files = src/res/io.jamulus.jamulus.svg src/res/io.jamulus.jamulusserver.png
```
Comment (pljones) [inline Jamulus.pro:344]: `src/res/io.jamulus.jamulusserver.png` should be `.svg` here, I think.

File: Jamulus.pro [hunk 3]
```diff
@@ -683,11 +683,11 @@ DISTFILES += ChangeLog \
...
-    distributions/jamulus.desktop.in \
-    distributions/jamulus-server.desktop.in \
-    distributions/jamulus.png \
-    distributions/jamulus.svg \
-    distributions/jamulus-server.svg \
+    linux/jamulus.desktop.in \
+    linux/jamulus-server.desktop.in \
+    src/res/io.jamulus.jamulus.png \
+    src/res/io.jamulus.jamulus.svg \
+    src/res/io.jamulus.jamulusserver.png \
```
Comment (pljones) [inline Jamulus.pro:690]: Same again - either it's `.svg` or the src/res name is wrong.

File: Jamulus.pro (hunk 2 above)
Comment (ann0see) [reply Jamulus.pro]: Fixed.

**Review by pljones — APPROVED** [review]
Looks like it's only the dependeny issue again, now.

**Comment by ann0see** [comment]
It looks ok on Debian too. 

@corrados should know that this Pull Request moves raspijamulus.sh to the linux/ folder. In future we might move it again, depending on the restructuring of the repo layout.

**Comment by corrados** [comment]
> @corrados should know that this Pull Request moves raspijamulus.sh to the linux/ folder.

Thanks for letting me know. It's totally fine with me :-).

**Review by hoffie — APPROVED** [review]
After careful diff scanning, this looks fine to me... :)

**Comment by ann0see** [comment]
Thanks @corrados for your OK.

**Pull request merged**